go/libkb/client.go

```diff
@@ -136,36 +136,43 @@ func NewClient(e *Env, config *ClientConfig, needCookie bool) *Client {
 		jar, _ = cookiejar.New(nil)
 	}
 
-	var xprt *http.Transport
-	var timeout time.Duration
+	// http.Transport is an example of Go's implicit default behavior causing
+	// trouble. The "DefaultTransport" has many non-zero limits set in it,
+	// while a "zero Transport" doesn't set any limits at all and also disables
+	// the ProxyFromEnvironment. We copy the global default rather than
+	// starting from zero.
+	var globalDefaultTransport *http.Transport = http.DefaultTransport.(*http.Transport)
```

Comment:
are we going to want to change defaults dynamically at runtime? On the basis of mobile vs. desktop, for instance?